internal/resolver/delegatingresolver/delegatingresolver_test.go

```diff
@@ -0,0 +1,306 @@
+		ReportErrorF: func(err error) { errCh <- err },
+	}
+	return tcc, stateCh, errCh
+}
+
+// TestParsedURLForProxyEnv verifies that the parsedURLForProxy function
+// correctly resolves the proxy URL for a given target address.
+func (s) TestParsedURLForProxyEnv(t *testing.T) {
+	// Overwrite the function in the test and restore them in defer.
+	hpfe := func(req *http.Request) (*url.URL, error) {
+		if req.URL.Host == targetTestAddr {
+			return &url.URL{
+				Scheme: "https",
+				Host:   envProxyAddr,
+			}, nil
+		}
+		return nil, nil
+	}
+	defer overwrite(hpfe)()
+
+	// envTestAddr should be handled by ProxyFromEnvironment.
+	got, err := parsedURLForProxy(targetTestAddr)
+	if err != nil {
+		t.Errorf("Unable to get proxy URL : %v\n", err)
+	}
+	if got.Host != envProxyAddr {
+		t.Errorf("want %v, got %v", envProxyAddr, got)
+	}
+}
+
+// TestDelegatingResolverNoProxy verifies that the delegating resolver correctly
+// sends the resolved target URI to the ClientConn.
+func (s) TestDelegatingResolverNoProxy(t *testing.T) {
+	mr := manual.NewBuilderWithScheme("test") // Set up a manual resolver to control the address resolution.
+	target := "test:///" + targetTestAddr
+
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	// Create a delegating resolver with no proxy configuration
+	dr, err := New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mr, false)
+	if err != nil || dr == nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	// Update the manual resolver with a test address.
+	mr.UpdateState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the same address.
+	expectedState := resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
+
+	if state := <-stateCh; len(state.Addresses) != 1 || !cmp.Equal(expectedState, state) {
+		t.Fatalf("Unexpected state from delegating resolver: %v, want %v", state, expectedState)
+	}
+}
+
+// setupDNS unregisters the DNS resolver and registers a manual resolver for the
+// same scheme. This allows the test to mock the DNS resolution for the proxy resolver.
+func setupDNS(t *testing.T) *manual.Resolver {
+
+	mr := manual.NewBuilderWithScheme("dns")
+
+	dnsResolverBuilder := resolver.Get("dns")
+	resolver.Register(mr)
+
+	t.Cleanup(func() { resolver.Register(dnsResolverBuilder) })
+	return mr
+}
+
+// TestDelegatingResolverwithDNSAndProxy verifies that the delegating resolver
+// correctly updates state with resolver proxy address with resolved target
+// URI as attribute of the proxy address when the target URI scheme is DNS and
+// a proxy is configured and target resolution is enabled.
+func (s) TestDelegatingResolverwithDNSAndProxyWithTargetResolution(t *testing.T) {
```

Comment:
Also, could you please consider moving these tests which test the delegating resolver's API to `delegatingresolver_ext_test.go`, and have the package name be `delegatingresolver_test` in there. This will ensure that these tests don't rely on any of the implementation details by mistake and instead only test the exported API.

Reply:
Done.